### ISSUE TITLE
[feature-request]: Enhance listtransactions rpc

### ISSUE THREAD

**Issue by niteshbalusu11** [ElementsProject/lightning#7397]
I was working on a transactions view in zeus for onchain and noticed that to get to get info on actual deposits and withdrawals, its really hard in CLN. I am having to work with 3 different RPCs to get accurate info on what the tx is for.
Have to do `listaccountevents` which is an expensive call because it returns a lot of stuff and also check against `listtransactions`.

A nice improvement to `listtransactions` rpc would be to add a `tag` and `type` to indicate what transaction it is so that we can easily get the actual info of the tx that's going out or coming in. 
It would also also extremely help if the outputs are tagged saying if its change coming back to our wallet or going to an external address.

This is just my thoughts on the rpc, open to other ideas if there are better ways of doing this.

**Comment by lnbc1QWFyb24** [comment]
Yeah it is definitely not easy to categorise onchain transactions with the API as it currently is, and I agree it would be nice if the inputs and outputs were tagged indicating if the wallet owns them. The `bkpr-listaccountevents` command also does not include events that happened before that plugin was added as far as I am aware. 

Another way you can approach it for now is to use the `listfunds` command with `spent=true` to get all utxos that the wallet knows about under the `outputs` param of the response. You can then match the utxos to onchain transactions from `listtransactions` and infer whether inputs were spent, outputs are change or receives etc. Again not the greatest dev experience, but doable.

**Comment by kaloudis** [comment]
This comment does a pretty good job at illustrating the issue at hand: https://github.com/ZeusLN/zeus/pull/2228#issuecomment-2166804428
